browser/ui/webui/brave_wallet/wallet_panel_ui_browsertest.cc

```diff
@@ -5,66 +5,295 @@
+using content::EvalJsResult;
+
+namespace {
+
+std::string SelectInNetworkList(const std::string& selector) {
+  return base::StringPrintf("window.testing.walletNetworks.querySelector(`%s`)",
+                            selector.c_str());
+}
+
+std::string SelectInAddNetworkDialog(const std::string& selector) {
+  return base::StringPrintf(
+      "window.testing.addWalletNetworkDialog.querySelector(`%s`)",
+      selector.c_str());
+}
+
+std::string DoubleClick() {
+  return ".dispatchEvent((function (){const e = "
+         "document.createEvent('MouseEvents');e.initEvent('dblclick',true,true)"
+         ";return e;})())";
+}
+
+std::string CeloNetwork() {
+  return "[data-test-chain-id='chain-0xa4ec']";
+}
+
+std::string PolygonNetwork() {
+  return "[data-test-chain-id='chain-0x89']";
+}
+
+std::string NetworkNameSpan() {
+  return "[class|='NetworkName']";
+}
+
+std::string CeloNetworkHideButton() {
+  return CeloNetwork() + " .hide-network-button";
+}
+
+std::string CeloNetworkChainName() {
+  return CeloNetwork() + " .chainName";
+}
+
+std::string NetworksButton() {
+  return R"([data-test-id='select-network-button'])";
+}
+
+std::string Select(const std::string& selector) {
+  return base::StringPrintf(R"(document.querySelector(`%s`))",
+                            selector.c_str());
+}
+
+std::string Select(const std::string& selector1, const std::string& selector2) {
+  return base::StringPrintf(R"(document.querySelector(`%s %s`))",
+                            selector1.c_str(), selector2.c_str());
+}
+
+void NonBlockingDelay(const base::TimeDelta& delay) {
+  base::RunLoop run_loop(base::RunLoop::Type::kNestableTasksAllowed);
+  base::ThreadTaskRunnerHandle::Get()->PostDelayedTask(
+      FROM_HERE, run_loop.QuitWhenIdleClosure(), delay);
+  run_loop.Run();
+}
+
+bool WaitFor(content::WebContents* web_contents, const std::string& selector) {
+  for (int i = 0; i < 1000; ++i) {
+    if (EvalJs(web_contents, "!!(" + selector + ")").ExtractBool())
+      return true;
+    NonBlockingDelay(base::Milliseconds(10));
   }
+  return false;
+}
+
+bool WaitAndClickElement(content::WebContents* web_contents,
+                         const std::string& selector) {
+  if (!WaitFor(web_contents, selector))
+    return false;
+
+  return EvalJs(web_contents, selector + ".click()").value.is_none();
+}
 
+bool ClickElementAsync(content::WebContents* web_contents,
+                       const std::string& selector) {
+  auto script =
+      base::StringPrintf("setTimeout(() =>{%s.click()}, 0)", selector.c_str());
+  return EvalJs(web_contents, script).value.is_none();
+}
+
+}  // namespace
+
+class WalletPanelUIBrowserTest : public InProcessBrowserTest {
+ public:
   void SetUpOnMainThread() override {
-    webui_contents_ = content::WebContents::Create(
-        content::WebContents::CreateParams(browser()->profile()));
+    InProcessBrowserTest::SetUpOnMainThread();
+
+    BraveSettingsUI::DisableCSPForTesting() = true;
```

Comment:
Please document in the code for why disabling CSP is needed.

Reply:
https://github.com/brave/brave-core/pull/13792/commits/c5bde88900b3ac8c703bae016e62c3816e6ad64e